server/controllers/messageController.js

```diff
@@ -18,16 +18,24 @@ exports.create = async (req, res, next) => {
 
 exports.read = async (req, res, next) => {
   try {
-    const page = 30 * req.query.page;
+    const limit = req.query.size * 1;
+    const page = limit * req.query.page;
     const messages = await Message.find({
       conversation: req.params.convoID,
     })
-      .sort({ dateCreated: -1 })
+      .sort({ _id: -1 })
       .skip(page)
-      .limit(30);
-    res.status(200).json({
-      messages,
-    });
+      .limit(limit)
+      .populate("author");
+    if (messages.length > 0) {
+      res.status(200).json({
+        messages: messages.reverse(),
+      });
```

Comment:
Aren't the messages already being returned in reverse sorted order?

Reply:
Yeah maybe this is silly, all my rendering functions are [oldest -> newest]. I could remove this reversal and change the rendering functions to render in reverse order and shift new messages rather than push.